keanu-project/src/main/java/io/improbable/keanu/tensor/bool/SimpleBooleanTensor.java

```diff
@@ -46,9 +47,9 @@ public SimpleBooleanTensor(int[] shape) {
 
     public SimpleBooleanTensor(boolean constant, int[] shape) {
         this.data = new boolean[(int) TensorShape.getLength(shape)];
+        Arrays.fill(this.data, constant);
```

Comment:
What's the difference between `System.arraycopy` and `Arrays.fill`? Why are you using both in this class?

Reply:
Fill is for filling a single value into a data buffer for an arbitrary shape e.g. a 2x2 of trues. Array copy is used to copy from the user provided boolean buffer to an internal buffer. This copy protects against the case where you pass in more data than the shape allows.